vowpalwabbit/test_common/include/vw/test_common/matchers.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+MATCHER(ActionScoreEqual, "")
+{
+  auto lhs_action = std::get<0>(arg).action;
+  auto rhs_action = std::get<1>(arg).action;
+  testing::Matcher<uint32_t> action_matcher = testing::Eq(lhs_action);
+
+  auto lhs_score = std::get<0>(arg).score;
+  auto rhs_score = std::get<1>(arg).score;
+  testing::Matcher<float> score_matcher = testing::FloatEq(lhs_score);
```

Comment:
Near takes a tolerance, float eq checks they are approximately equal. In my testing this has proven to work well and remove the need for us to define a tolerance.

Reply:
ok lets see how it goes

Reply:
It will be really easy to add a tolerance to this matcher if we need it. We change `MATCHER` to `MATCHER_P` and it takes an arg we can use as tolerance